core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -102,6 +106,26 @@ public static void dropTableData(FileIO io, TableMetadata metadata) {
       deleteFiles(io, manifestsToDelete);
     }
 
+    if (io instanceof SupportsBulkOperations) {
```

Comment:
I feel like there are three types of deletes. What about defining helper methods for them that would abstract some common logic like try catch? The type annotation is optional, it can probably be deduced from the file name.

```
private static void delete(FileIO io, String file, String type) {
  try {
    io.deleteFile(file);
  } catch (RuntimeException e) {
    LOG.warn("Failed to delete {}: {}", type, file, e);
  }
}

private static void delete(FileIO io, Iterable<String> files, String type) {
  Tasks.foreach(files)
      .executeWith(ThreadPools.getWorkerPool())
      .noRetry()
      .suppressFailureWhenFinished()
      .onFailure((file, exc) -> LOG.warn("Delete failed for {}: {}", type, file, exc))
      .run(io::deleteFile);
}

private static void bulkDelete(SupportsBulkOperations io, Iterable<String> files, String type) {
  try {
    io.deleteFiles(files);
  } catch (RuntimeException e) {
    LOG.warn("Failed to bulk delete {}", type, e);
  }
}
```

Reply:
Sure this makes a lot of sense! I originally left it as is to minimize the changes but this is a good opportunity to refactor especially since the logic for concurrently deleting manifest, metadata and manifest lists are all the same, just with different error messages. I've updated